### PR TITLE
Hrcpp 430/exec test

### PR DESCRIPTION
**Pull request by rigazilla**
This enables the swig test suite to run ExecTest.java class.

https://issues.jboss.org/browse/HRCPP-430

### CODE REVIEW

**Comment by alanfx** [comment]
Hey @rigazilla I am reviewing the changes now, but I getting one failure while running the tests:

`      Start 18: simpleSasl

18: Test command: /artifacts/cpp-client/build/simpleSasl
18: Test timeout computed to be: 1500
18: ERROR [TcpTransportFactory.cpp:263] Initial ping has thrown an exception when pinging 127.0.0.1:11222 : Sasl error -4 : starting SASL negotiation
18: ERROR [RetryOnFailureOperation.h:84] Exception encountered, retry 9 of 10: Sasl error -4 : starting SASL negotiation
18: terminate called after throwing an instance of 'infinispan::hotrod::HotRodClientException'
18:   what():  Sasl error -4 : starting SASL negotiation
18/57 Test #18: simpleSasl .......................***Exception: Other  0.13 sec`

File: jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java [hunk 1]
```diff
@@ -626,5 +632,34 @@ public ServerStatistics stats() {
...
     }
 
+    @Override
+    public <T> T execute(String scriptName, Map<String, ?> params) {
+        MapType argsMap = new MapType();
+        JBossMarshaller jb = new JBossMarshaller();
+        for (String parName : params.keySet()) {
+            ByteBuffer b;
+            try {
+                b= jb.objectToBuffer(params.get(parName));
```
Comment (alanfx) [inline jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java]: nit: space after the `b`. I just use the same IDE formatting rules as Infinispan Java code to fix these.

File: jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java [hunk 2]
```diff
@@ -626,5 +632,34 @@ public ServerStatistics stats() {
...
+                argsMap.set(parName, new String(b.getBuf()));
+            } catch (IOException | InterruptedException e) {
+                // TODO Auto-generated catch block
```
Comment (alanfx) [inline jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java]: Remove?

File: jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java [hunk 3]
```diff
@@ -626,5 +632,34 @@ public ServerStatistics stats() {
...
+        UCharVector executeResult = jniRemoteCache.execute(scriptName, argsMap);
+        byte[] buf = new byte[(int)executeResult.size()];
+        for (int i=0; i< buf.length; i++)
+        {
+            buf[i]= (byte)executeResult.get(i);
+        }
+        try {
+            return (T) jb.objectFromByteBuffer(buf);
+        } catch (ClassNotFoundException | IOException e) {
+            // TODO Auto-generated catch block
```
Comment (alanfx) [inline jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java]: Remove?

File: jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java [hunk 4]
```diff
@@ -626,5 +632,34 @@ public ServerStatistics stats() {
...
+            } catch (IOException | InterruptedException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+        UCharVector executeResult = jniRemoteCache.execute(scriptName, argsMap);
+        byte[] buf = new byte[(int)executeResult.size()];
+        for (int i=0; i< buf.length; i++)
+        {
+            buf[i]= (byte)executeResult.get(i);
```
Comment (alanfx) [inline jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java]: nit: Space before `=` and `<` in the `for` loop

**Comment by rigazilla** [comment]
@alanfx about simpleSasl: check if you have the cyrus plugin installed:

cyrus-sasl.x86_64                      2.1.26-32.fc26              @@commandline
cyrus-sasl-devel.x86_64                2.1.26-32.fc26              @@commandline
cyrus-sasl-gssapi.x86_64               2.1.26-32.fc26              @@commandline
cyrus-sasl-lib.x86_64                  2.1.26-32.fc26              @@commandline
cyrus-sasl-md5.x86_64                  2.1.26-32.fc26              @@commandline
cyrus-sasl-plain.x86_64                2.1.26-32.fc26              @@commandline
cyrus-sasl-scram.x86_64                2.1.26-32.fc26              @@commandline
✔ ~/git/cpp-client [HRCPP-430/ExecTest L|✚ 1…667⚑ 28]

**Comment by alanfx** [comment]
@rigazilla OK, I don't have all of those plugins installed. I'll get them and try again. The README should include these dependencies. (hint, hint!)

File: jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java (hunk 3 above)
Comment (rigazilla) [reply jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java]: the exceptions need to be caught to respect the method signature. Re throwing as runtime exception

Comment (alanfx) [reply jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java]: I was referring to removing the `TODO` comment

Comment (rigazilla) [reply jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java]: yeah, I've realized too late :D

**Comment by alanfx** [comment]
@rigazilla LGTM. Merging...

**Pull request merged**